docs/integration.md

```diff
@@ -7,110 +7,126 @@ It's recommended that you encapsulate integration code in a component or a helpe
-                            //Set the value to the selected option
-							ctrl.data.map(function(d){
-								if(d.id == id) {
-									ctrl.value(d);
-								}
-							});
-
-							if (typeof ctrl.onchange == "function"){
-								ctrl.onchange(el.select2("val"));
-							}
-							m.endComputation();
-						});
-				}
-				el.val(ctrl.value().id).trigger("change");
-			} else {
-				console.warn('ERROR: You need jquery and Select2 in the page');	
-			}
-		};
-	}
-};
-
-var Dashboard = {
-    controller: function() {
-        var ctrl = this,
-          //list of users to show
-          data = [
-            {id: 1, name: "John"}, 
-            {id: 2, name: "Mary"}, 
-            {id: 3, name: "Seniqua"}
-          ];
-      
-        ctrl.data = data;
-        //  Has to use a prop for the current user
-        ctrl.currentUser = m.prop(data[1]);
-        ctrl.changeUser = function(id) {
-          console.log(id);
-        };
-    },
-
-    view: function(ctrl) {
-        return m("div", [
-            m("label", "User:"),
-            m.component(Select2, {
-              data: ctrl.data, 
-              value: ctrl.currentUser, 
-              onchange: ctrl.changeUser
-            })
-        ]);
+// Component containing a Select that uses Select2 for functionality.
+var S2Component = {
+
+  controller: function() {
+    // this controller does nothing.
+    // but I like to include it for
+    // consistency in components.
+  },
+
+  // Rendered view for S2
+  view: function(ctrl, attrs) {
+    var current = attrs.selectedUser;
+
+    return m('select', {
+        class: 'select-field',
+        config: S2Component._configure
+      },
+      attrs.data.map(function(item){
+        return m('option', {
+                  id: item.id,
+                  value: item.value,
+                  selected: (item.id === current) ? true : false
+                }, item.name);
+      })
+    )
+  },
+
+   // Configure function -- called from m('select') in the view
+  _configure: function(element, initialized) {
+
+    /*
+      Note: This is a use of the config attribute given to
+      attributes on mithril objects. Integrating with 3rd
+      party DOM manipulation (jQuery) needs {config: function(){}} because
+      that's the attribute that exposes the real DOM element (as opposed to
+      the virtual DOM element) in the corresponding function so you can access
+      and manipulate it.
+
+      the function at the config property in attrs is passed the real DOM element
+      (element), and whether or not it exists in the page (initialized)
+
+      Anyways, let's look at using Select2 now.
+    */
+
+
+    // Ensure Externals exist, otherwise we'll error out.
+    if(!(jQuery && jQuery.fn.select2))
+      return console.error('You need jquery and Select2 in the page')
+
+    // jQuery element
+    var lmnt = $(element);
+
+    // If this hasn't been initialized, we can do our setup  
+    if(!initialized) {
+
+       lmnt.select2({
```

Comment:
Try using this instead:

```js
if (!initialized) {
  $(element).select2({
    // whatever
  })
}
```

It's a little more idiomatic for jQuery, since it's a one-time use.